File: identhub/src/main/java/de/solarisbank/identhub/di/LibraryModule.java
package de.solarisbank.identhub.di;

import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.RestrictTo;
import androidx.lifecycle.SavedStateHandle;
import androidx.lifecycle.ViewModel;

import java.util.Map;

import de.solarisbank.identhub.AssistedViewModelFactory;
import de.solarisbank.identhub.di.internal.Factory2;
import de.solarisbank.identhub.di.internal.Provider;

@RestrictTo(RestrictTo.Scope.LIBRARY)
public final class LibraryModule {

    private final Application application;

    public LibraryModule(Application application) {
        this.application = application;
    }

    public Context provideApplicationContext() {
        return application;
    }

    @NonNull
    public AssistedViewModelFactory provideViewModelFactory(Map<Class<? extends ViewModel>, Provider<ViewModel>> classProviderMap,
                                                            Map<Class<? extends ViewModel>, Factory2<ViewModel, SavedStateHandle>> classSavedStateProviderMap) {
        return new AssistedViewModelFactory(classSavedStateProviderMap, classProviderMap);
    }
}

File: core/src/main/java/de/solarisbank/sdk/feature/di/LibraryModule.kt
package de.solarisbank.sdk.feature.di

import android.app.Application
import android.content.Context
import androidx.annotation.RestrictTo
import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel
import de.solarisbank.sdk.feature.di.internal.Factory2
import de.solarisbank.sdk.feature.di.internal.Provider
import de.solarisbank.sdk.feature.viewmodel.AssistedViewModelFactory

@RestrictTo(RestrictTo.Scope.LIBRARY)
class LibraryModule(private val application: Application) {
    fun provideApplicationContext(): Context {
        return application
    }

    fun provideViewModelFactory(classProviderMap: Map<Class<out ViewModel>, Provider<ViewModel>>,
                                classSavedStateProviderMap: Map<Class<out ViewModel>, Factory2<ViewModel, SavedStateHandle>>): AssistedViewModelFactory {
        return AssistedViewModelFactory(classSavedStateProviderMap, classProviderMap)
    }
}